libavutil/sub_frame_metadata.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef AVUTIL_SUB_FRAME_METADATA_H
+#define AVUTIL_SUB_FRAME_METADATA_H
+
+#include "frame.h"
+
+/**
+ * Allocate a AVFrame and add it to the frame.
+ *
+ * @param frame The frame which side data is added to.
+ *
+ * @return The AVFrame structure to be filled by caller.
+ */
+AVFrame *av_sub_frame_create_side_data(AVFrame *frame);
```

Comment:
Who'll be responsible for freeing the this frame ? I think it should be documented